src/geometry/euclidean/triangle.lean

```diff
@@ -352,4 +352,47 @@ begin
                                                 (λ he, h2 (vsub_eq_zero_iff_eq.1 he))
 end
 
+/-- Stewart's Theorem. -/
+theorem sq_dist_mul_dist_add_sq_dist_mul_dist (a : P) {b c p : P} (h : ∠ b p c = π) :
+  (dist a b) ^ 2 * (dist c p) + (dist a c) ^ 2 * (dist b p) =
+  (dist b c) * ((dist a p) ^ 2 + (dist b p) * (dist c p)) :=
+begin
+  let lawcos := dist_sq_eq_dist_sq_add_dist_sq_sub_two_mul_dist_mul_dist_mul_cos_angle,
```

Comment:
Should we add an alias for this lemma?

Reply:
I favor that. I don't know how to add an alias though.

Reply:
Search for the word `alias` in mathlib, and you'll find some examples.